src/js/veteran-id-card/containers/Main.jsx

```diff
@@ -11,15 +11,36 @@ class Main extends React.Component {
     super(props);
     this.handleSubmit = this.handleSubmit.bind(this);
     this.renderButton = this.renderButton.bind(this);
+    this.renderVicForm = this.renderVicForm.bind(this);
+  }
+
+  componentDidUpdate() {
+    if (this.props.vicUrl) {
```

Comment:
Does the presence of certain traits affect whether we want to submit the request? That can be caught here too.

Reply:
Not so far...we do the "is veteran, has edipi" enforcement on the backend and will return an error code accordingly. Other business logic (like making sure honorable discharge) is all responsibility of VIC team so far.